### DISCUSSION TITLE
auto upload to cloud every time gopro stops recording

### DISCUSSION THREAD

**Discussion by Fougl** [gopro/labs#512] [Feature Requests]
Hi, I would like to know if this feature is available, and where would be the best location to requesst it? Best

**Comment by dnewman-gpsw** [comment]
If on HERO11, you could create a script to do this.

**Reply by Fougl** [reply]
Really? to cloud like gopro subscription cloud?

**Reply by dnewman-gpsw** [reply]
Yes, the !U command starts an upload the GoPro Cloud with your subscription.  Users use this for end of day timelapse uploads, but it could be scripted so that any end of capture would start an upload.

**Reply by Fougl** [reply]
Amazing. No such function on Gopro Max? Would it be possible?

**Reply by dnewman-gpsw** [reply]
!U just enables the existing function in the camera, but unfortunately it is not in Max.

**Reply by Fougl** [reply]
Max doesnt have auto upload function through quik?

**Comment by dnewman-gpsw** [comment]
Not sure, it might only be for HERO mode captures

**Reply by Fougl** [reply]
https://community.gopro.com/s/article/Auto-upload-your-footage-to-the-cloud?language=en_US

![image](https://github.com/gopro/labs/assets/21178929/ab6b3992-8bcf-4f35-afe9-b02189bc22a0)

Does this mean that Max could also use U!? WOuld it already work, or labs firmware would need to be changed? This would save my life 10 times

**Reply by dnewman-gpsw** [reply]
Best way to know, try it.

**Reply by Fougl** [reply]
Ok, I will. However, i just remembered max does not have boot commands so it doesnt help me...
Returning to the hero 11: would keeping the connection to the wifi network with W help to speed up the auto uploads since connecting everytime is not necessary? It actually takes really long time sometimes to initiate the upload. If yes or to test it, I need to ask how to add the JOIN metadata?

**Reply by dnewman-gpsw** [reply]
MAX should have boot commands, it was updated a few months ago.

Experiment on H11, as I don't remember if there are ways to speed it up.

**Reply by Fougl** [reply]
ah cool. you should update it in the table in the main page. if it has boot command, I need to ask how to start auto upload everytime the button to stop recording gets pressed. I was searching but only way is that i save into a variable if camera is recording and when the value switches from 1 to 0, to trigger the upload. but it seems complicated

**Reply by Fougl** [reply]
Hi, this is the update. I tested the following command SM3D4H5!U!R

Hero 11: works perfectly
Hero 11 mini: doesnt work, it just repeats the motion detection, no upload occurs. Can you confirm? Any plans to repair the bug?
Hero 10: works, but after the upload you have to touch the screen with your hand, than repeats the motion detection + upload; Can you confirm? any plans to repair the bug?
![image](https://github.com/gopro/labs/assets/21178929/a0cb9e2f-2392-49b9-a433-e783caa4e1fb)

MAX: same happens with max as hero 10. Can you confirm? any plans to repair the bug?

Also, Boot command !MBOOT="!Ltest"!SAVEtest=!SM3D4H5!R is not working on MAX, can you confirm? Does this mean tha boot command is not working on max?

**Comment by Fougl** [comment]
Hi, just checkin if I can expect an update on this one. Would be unbelievably helpful if auto upload and boot could be working an all cameras. Best

**Reply by dnewman-gpsw** [reply]
Which camera?  It might happen, but slammed with HERO12, and other updates.

**Reply by Fougl** [reply]
this is the reference reply https://github.com/gopro/labs/discussions/512#discussioncomment-6930963

**Reply by dnewman-gpsw** [reply]
This topic has drifted that I now don't know what you are asking for.

**Reply by Fougl** [reply]
Well, its easier if you read all the previous 2 or 3 replies than if I explain again. Or not?

**Reply by dnewman-gpsw** [reply]
Make it simple, and be precise.

**Reply by Fougl** [reply]
Well, im trying to use the U! command on all cameras, with motion detection with R! Work properly only with HERO 11 Black (all the details from my tests are here https://github.com/gopro/labs/discussions/512#discussioncomment-6930963
I am wondering if anytime this fixes can be expected. Especially for MAX, I would be grateful

Also, you said that boot commands for MAX were added few months ago. For the latest firmware posted in April, this is not the case. I am wondering

**Reply by dnewman-gpsw** [reply]
Unlikely on MAX, maybe on Mini. 

MAX works with boot commands, I just tested and it worked as documented.
